src/platforms/evdev/libinput_device.cpp

```diff
@@ -92,36 +92,52 @@ auto get_scroll_axis(libinput_event_pointer* event, libinput_pointer_axis axis,
         return {};
     }
 
+#ifdef MIR_LIBINPUT_HAS_VALUE120
+    auto const libinput_event_type = libinput_event_get_type(libinput_event_pointer_get_base_event(event));
+
+    mir::geometry::generic::Value<float, Tag> const precise{
+        libinput_event_pointer_get_scroll_value(event, axis) * scale};
+    auto const stop = precise.as_value() == 0;
+    mir::geometry::generic::Value<int, Tag> const discrete{0};
```

Comment:
In the current `ScrollAxis` design we need to provide both `discrete` and `value120` regardless of libinput version. Older Wayland clients will be sent the discrete value and newer ones the value120. We could store only value120  in the event, but this would require all consumers that want discrete events (including the Wayland frontend and possibly shells) to have their own accumulators.

Reply:
Ack. That's one of the things I didn't get to checking. I'm hoping it will all be fixed before I'm back at work. ;)